### PR TITLE
Validator state

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Adds validator state that can persist validator information and implements `uptime.State`.

## How this was tested

Added UT

## How is this documented

Comments in code

### CODE REVIEW

File: plugin/evm/validators/state_test.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	expectedStartTime := time.Now()
+
+	// add listener
+	listener := &testCallbackListener{
+		t: t,
+		onAdd: func(vID ids.ID, nodeID ids.NodeID, startTime uint64, isActive bool) {
+			require.Equal(expectedvID, vID)
+			require.Equal(expectedNodeID, nodeID)
+			require.Equal(uint64(expectedStartTime.Unix()), startTime)
+			require.True(isActive)
```
Comment (darioush) [inline plugin/evm/validators/state_test.go:231]: should probably have some verification that these callbacks are actually invoked

**Review by darioush — APPROVED** [review]

File: plugin/evm/validators/state_test.go (hunk 1 above)
Comment (ceyonur) [reply plugin/evm/validators/state_test.go]: dunno what I was thinking, mocks are much easier...

File: plugin/evm/validators/state_test.go [hunk 2]
```diff
@@ -216,71 +217,34 @@ func TestStateListener(t *testing.T) {
 	db := memdb.New()
 	state, err := NewState(db)
 	require.NoError(err)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```
Comment (darioush) [inline plugin/evm/validators/state_test.go:221]: nit: I think calls to Finish are not required in recent gomock versions.

**Review by darioush — APPROVED** [review]

File: plugin/evm/validators/state_test.go (hunk 2 above)
Comment (ceyonur) [reply plugin/evm/validators/state_test.go:221]: let's keep it for now

**Pull request merged**